### DISCUSSION TITLE
Checkbox handler

### DISCUSSION THREAD

**Discussion by BesnikMahmuti-Prosperoware** [codaxy/cxjs#669] [General] [answered]
Hi is there a way to get the selected checkbox record so that we know the object with the details of a selected or unselected checkbox, for example like we have onClick event for buttons.

**Comment by mstijak** [comment]
If checkboxes are rendered in a Repeater you can iterate the underlying array and filter out unchecked records. There is no callback at the moment, but we can add it if there is a valid use-case.

**Reply by BesnikMahmuti-Prosperoware** [reply]
In my case, I have a tree grid with nodes and each node has a checkbox next to it when I select one I want to get that node object. I know that one solution is adding a trigger for $record.checked. But wanted to know if there is a better way for this.

**Comment by mstijak** [comment, marked as answer]
You can access the store in the set handler.

```js
<Checkbox
      value={{
        bind: "$record.selected",
        set: (value, { store }) => {
          let record = store.get("$record"); //access to the record
          store.set("$record.selected", value);
        }
      }}
    >
      Test
    </Checkbox>
```

**Reply by BesnikMahmuti-Prosperoware** [reply]
Thank you very much Marko.
